src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/NumberHandlingTests.cs

```diff
@@ -88,13 +88,17 @@ private static void RunAsRootTypeTest<T>(List<T> numbers)
 
         private static string GetNumberAsString<T>(T number)
         {
-            return number switch
+            switch (Type.GetTypeCode(typeof(T)))
```

Comment:
please add a comment why original code couldn't be used so that we're not tempted of refactoring this back. Btw. isn't this implementation problem? It seems to me both codes are equivalent

Reply:
There is some issue with original code switch it doesn't work correct for floats on x86 android. I opened issue for that https://github.com/dotnet/runtime/issues/68906